34-chinese-pangolin/technical_specification.md

```diff
@@ -0,0 +1,83 @@
+# File Deletion and Outbox Cache Strategy (Chinese Pangolin)
+**Epic Type:** Implementation Epic
+
+## Scope
+### Outline:
+
+This Epic adds a new service, which provides a deletion endpoint for files.
+Calling this enpoint internally publishes an event.
+All services subscribing to this event will then delete this file from their databases and from S3 buckets they have write access to.
+
+### Included/Required:
+
+This Epic includes:
+
+- The creation of a new service, the **Purge Controller Service (PCS)**, which provides a RESTfull HTTP DELETE enpoint /files/{file_id}.
+- Upon receiving and validating the deletion request, it publishes an event
+- The following services subscribe to the new event and do the following after consuming:
+    - IFRS: Deletes the file from permanent storage (IRS would be better, but IRS does not have the connection between outward-facing file_id and S3 file name)
+    - IFRS: Send API Call to EKSS to delete the secret
+    - IFRS: Delete file from outbox
+    - IFRS: Delete file entry from its database
+    - DCS: Delete file entry from its database
+    - Both services: Publish confirmation event afterwards
+- The EKSS provides a new RESTful HTTP DELETE endpoint /sercets/{secret_id} that deletes the corresponding secret from vault.
+
+These parts of the epic are described in detail below.
+
+### Optional: Caching Strategy for DCS/Outbox
+
+- Add a "last_accessed" field for each entry in the DCS files database
+- Each time, this file is sucessfully accessed via a download request, update the "last_accessed" field for this files entry
+- Provide Script that looks through all files currently in the Outbox. If their "last_accessed" time is longer ago then a certain threshold, delete that file.
```

Comment:
```suggestion
- Provide a script that scans the "last_accessed" field of all files currently present in the Outbox. If the timedelta since the last access is above a certain (to be determined) threshold, delete the respective file from the Outbox.
```
Phrasing with regard to file deletion is ambiguous here, even if intent is clear